File: backend/src/main/java/com/saas_learning/backend/entities/Transaction.java
package com.saas_learning.backend.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Table(name = "transactions")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Transaction {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private LocalDate date;

    private String description;

    private Double montant;

    private String categorie;

    @ManyToOne
    private User user; // pour lier les transactions à un utilisateur

}

File: front-saas-training/types/transaction.ts
import type { User } from "./user";

export interface Transaction {
  id: number;
  date: string; // Ou Date si vous transformez la string en Date côté front
  description: string;
  montant: number;
  categorie: string;
  user: User | number; // Selon si vous recevez l'objet complet ou juste l'ID
}
